pkg/apis/v1alpha1/serviceexport.go

```diff
@@ -88,3 +92,133 @@ type ServiceExportList struct {
 	// +listType=set
 	Items []ServiceExport `json:"items"`
 }
+
+// ServiceExportConditionType is a type of condition associated with ServiceExport.
+// This type should be used with the ServiceExportStatus.Conditions
+// field.
+type ServiceExportConditionType string
+
+// ServiceExportConditionReason defines the set of reasons that explain
+// why a particular conflict condition type has been raised.
+type ServiceExportConditionReason string
+
+const (
+	// ServiceExportConditionValid is true when the Service Export is valid. This does
+	// not indicate whether or not the configuration has been exported to
+	// a control plane / data plane.
+	//
+	//
+	// Possible reasons for this condition to be true are:
+	//
+	// * "Valid"
+	//
+	// Possible reasons for this condition to be False are:
+	//
+	// * "NoService"
+	// * "InvalidServiceType"
+	//
+	// Controllers may raise this condition with other reasons,
+	// but should prefer to use the reasons listed above to improve
+	// interoperability.
+	ServiceExportConditionValid ServiceExportConditionType = "Valid"
+
+	// ServiceExportReasonValid is used with the "Valid" condition when the
+	// condition is True.
+	ServiceExportReasonValid ServiceExportConditionReason = "Valid"
+
+	// ServiceExportReasonNoService is used with the "Valid" condition when
+	// the associated Service does not exist.
+	ServiceExportReasonNoService ServiceExportConditionReason = "NoService"
+
+	// ServiceExportReasonInvalidServiceType is used with the "Valid"
+	// condition when the associated Service has an invalid type
+	// (per the KEP at least the ExternalName type).
+	ServiceExportReasonInvalidServiceType ServiceExportConditionReason = "InvalidServiceType"
+)
+
+const (
+	// ServiceExportConditionExported is true when the service is exported to some
+	// control plane or data plane.
+	//
+	//
+	// Possible reasons for this condition to be true are:
+	//
+	// * "Exported"
+	//
+	// Possible reasons for this condition to be False are:
+	//
+	// * "Invalid"
+	// * "Pending"
+	//
```

Comment:
I think we need a `Failed` reason as well to reflect an error that prevented export (eg API server not available).

Reply:
I'm wondering if `Pending` wouldn't work in this case? Also not sure if this should be `Failing` or `Failed`.

Reply:
I don't think so. `Pending` means we haven't tried it yet.

Reply:
Sounds good, I will add that!